supabase/functions/billing/index.ts

```diff
@@ -0,0 +1,74 @@
+import { serve } from "https://deno.land/std@0.184.0/http/server.ts";
+import { corsHeaders } from "../_shared/cors.ts";
+import { setupIntent } from "./setup_intent.ts";
+import { getTenantPaymentMethods } from "./get_tenant_payment_methods.ts";
+import { deleteTenantPaymentMethod } from "./delete_tenant_payment_method.ts";
+import { setTenantPrimaryPaymentMethod } from "./set_tenant_primary_payment_method.ts";
+import { createClient } from "https://esm.sh/@supabase/supabase-js@2.0.5";
+
+serve(async (req) => {
+    let res: ConstructorParameters<typeof Response> = [null, {}];
+    try {
+        if (req.method === "OPTIONS") {
+            res = ["ok", { status: 200 }];
```

Comment:
This cors handling threw me for a bit of a loop because I'd expected to see the cors headers set on this line. Is there a reason that we need to add the cors headers to _every_ response down on line 69?

Reply:
Looks like the headers need to be on both the preflight response and all actual responses: https://stackoverflow.com/questions/24264574/cors-headers-present-only-on-preflight-or-every-request